src/client/extensionActivation.ts

```diff
@@ -131,6 +131,7 @@ async function activateLegacy(
 
     const manager = serviceContainer.get<IExtensionActivationManager>(IExtensionActivationManager);
     context.subscriptions.push(manager);
+    manager.activateSync();
     const activationPromise = manager.activate();
```

Comment:
Is this necessary if you're activating sync?

Reply:
Yes, this guarantees the fact that it will happen before we execute the rest of the async code.
We need to ensure some of the code has been setup before we can start async code.
E.g. setting of shared properties, initiallizing some stuff before we start async, async stuff could be things like `starting juypyter` in the background based on auto start.